src/content/polyfill.ts

```diff
@@ -0,0 +1,123 @@
+        }
+      }
+      if (val == null /* OR undefined*/) {
+        handlers.delete(this)
+        removeAnyExisting()
+      } else if (typeof val === 'function') {
+        removeAnyExisting()
+        this.addEventListener('monetization', val)
+        handlers.set(this, val)
+      } else {
+        throw new Error('val must be a function, got ' + typeof val)
+      }
+    }
+  }
+
+  const supportsOriginal = DOMTokenList.prototype.supports
+  const supportsMonetization = Symbol.for('link-supports-monetization')
+  DOMTokenList.prototype.supports = function (token) {
+    if (this[supportsMonetization] && token === 'monetization') {
+      return true
+    } else {
+      return supportsOriginal.call(this, token)
+    }
+  }
+
+  const relList = Object.getOwnPropertyDescriptor(
+    HTMLLinkElement.prototype,
+    'relList'
+  )
+  const relListGetOriginal = relList.get
+
+  relList.get = function () {
+    const val = relListGetOriginal.call(this)
+    val[supportsMonetization] = true
+    return val
+  }
+
+  Object.defineProperty(HTMLLinkElement.prototype, 'relList', relList)
+  Object.defineProperty(HTMLElement.prototype, 'onmonetization', attributes)
+  Object.defineProperty(Window.prototype, 'onmonetization', attributes)
+  Object.defineProperty(Document.prototype, 'onmonetization', attributes)
+
+  let eventDetailDeprecationEmitted = false
+  class MonetizationEvent extends Event {
+    public readonly amountSent: PaymentCurrencyAmount
+    public readonly incomingPayment: string
+    public readonly paymentPointer: string
+
+    constructor(details) {
+      super('monetization', { bubbles: true })
+      const { amountSent, incomingPayment, paymentPointer } = details
+      this.amountSent = amountSent
+      this.incomingPayment = incomingPayment
+      this.paymentPointer = paymentPointer
+    }
+
+    get [Symbol.toStringTag]() {
+      return 'MonetizationEvent'
+    }
+
+    get detail() {
+      if (!eventDetailDeprecationEmitted) {
+        const msg = `MonetizationEvent.detail is deprecated. Access attributes directly instead.`
+        // eslint-disable-next-line no-console
+        console.warn(msg)
+        eventDetailDeprecationEmitted = true
+      }
+      const { amountSent, incomingPayment, paymentPointer } = this
+      return { amountSent, incomingPayment, paymentPointer }
+    }
+  }
+
+  window.MonetizationEvent = MonetizationEvent
+
+  window.addEventListener(
+    '__wm_ext_monetization',
+    (event) => {
+      if (!(event.target instanceof HTMLLinkElement)) return
+      if (!event.target.isConnected) return
+
+      const monetizationTag = event.target
+      monetizationTag.dispatchEvent(new MonetizationEvent(event.detail))
+    },
+    { capture: true }
+  )
+
+  window.addEventListener(
+    'onmonetization-attr-changed',
+    (event) => {
+      // eslint-disable-next-line no-console
+      console.log('onmonetization-attr-changed', event.detail)
```

Comment:
I feel like we should not log this. If we want to keep it, let's make it a debug log.

Reply:
Why is this event even fired though? I just left it as is for now.

Reply:
I'm yet to check how this event supposed to work. Will do within this PR then.

Reply:
It is used to register the `onmonetization` event handler for an element. Every time the handler is changed, we need to re-register the function.

```html
<link rel="monetization" ... onmonetization="console.log(event)" />
```

- When the page is loaded, we register the `onmonetization` event handler;
- If the website is changing the handler, we need to override the previous one;

Reply:
Removed log.
Fixed bug: make sure event is fired and caught by polyfill (timing issue)
Refactor: renamed the internal event to start with `__wm_ext`.